doc/source/user_guide/numpy_images.rst

```diff
@@ -168,43 +167,43 @@ These conventions are summarized below:
   =========================   ========================================
 
 
-Many functions in scikit-image operate on 3D images directly:
+Many functions in ``scikit-image`` can operate on 3D images directly:
 
     >>> im3d = np.random.rand(100, 1000, 1000)
     >>> from skimage import morphology
     >>> from scipy import ndimage as ndi
     >>> seeds = ndi.label(im3d < 0.1)[0]
     >>> ws = morphology.watershed(im3d, seeds)
 
-In many cases,
-the third imaging dimension has lower resolution than the other two.
-Some scikit-image functions provide a ``spacing`` keyword argument
-to process these images:
+In many cases, however, the third spatial dimension has lower resolution
+than the other two. Some ``scikit-image`` functions provide a ``spacing``
+keyword argument to help handling this situation:
 
     >>> from skimage import segmentation
     >>> slics = segmentation.slic(im3d, spacing=[5, 1, 1], multichannel=False)
 
-
-Other times, processing must be done plane-wise. When planes are the
-leading dimension, we can use the following syntax:
+Other functions may be lacking support for any non-2D images, and have to be used
+for processing of such data in plane-wise manner. When planes are stacked
+along the leading dimension (in agreement with our convention), the following
+syntax can be used:
 
     >>> from skimage import filters
-    >>> edges = np.zeros_like(im3d)
+    >>> edges = np.empty_like(im3d)
     >>> for pln, image in enumerate(im3d):
-    ...     # iterate over the leading dimension (planes)
+    ...     # Iterate over the leading dimension 
     ...     edges[pln] = filters.sobel(image)
 
 
-Notes on array order
---------------------
+Notes on the order of array dimensions
+--------------------------------------
 
-Although the labeling of the axes seems arbitrary, it can have a
-significant effect on speed of operations. This is because modern
-processors never retrieve just one item from memory, but rather a
-whole chunk of adjacent items. (This is called prefetching.)
-Therefore, processing elements that are
-next to each other in memory is faster than processing them
-in a different order, even if the number of operations is the same:
+Although the labeling of the axes might seems arbitrary, it can have a
+significant effect on the speed of operations. To put it simply, this
+is related to the fact that the modern processors never retrieve just
```

Comment:
I think the original "this is because..." is superior to "this is related to the fact that..."